tests/SCF/test_SCF.py

```diff
@@ -1,19 +1,82 @@
 import pytest
 import SCF
+import pickle
+import numpy as np
+
+Suv = pickle.load(open("suv.pkl", "rb"))
+Tuv = pickle.load(open("tuv.pkl", "rb"))
+Vuv = pickle.load(open("vuv.pkl", "rb"))
+eri = pickle.load(open("eri.pkl", "rb"))
+Huv = SCF.calc_hcore_matrix(Tuv, Vuv)
 
 
 def test_calc_nuclear_repulsion_energy(mol_h2o):
-    assert True
-#    assert SCF.calc_nuclear_repulsion_energy(mol_h2o) == 8.00236706181077,\
-#        "Nuclear Repulsion Energy Test (H2O) Failed"
+    assert SCF.calc_nuclear_repulsion_energy(mol_h2o) == 8.00236706181077, \
+        "Nuclear Repulsion Energy Test (H2O) Failed"
+
+
+def test_calc_hcore_matrix(mol_h2o):
+    assert SCF.calc_hcore_matrix(Tuv, Vuv)[0, 0] == \
+        pytest.approx(-32.57739541261037) and \
+        SCF.calc_hcore_matrix(Tuv, Vuv)[3, 4] == 0.0 and \
+        SCF.calc_hcore_matrix(Tuv, Vuv)[4, 3] == 0.0, \
+        "Initial Hcore Matrix Test (H2O) Failed"
 
 
 def test_calc_initial_density(mol_h2o):
-    """
-    Tests that the initial density returns a zero matrix
-    and tests dimensions
-    """
+    assert np.sum(SCF.calc_initial_density(mol_h2o)) == 0, \
+        "Initial Density matrix Test (H2O) Failed"
+
+
+def test_calc_fock_matrix(mol_h2o):
+    Duv = SCF.calc_initial_density(mol_h2o)
+    assert SCF.calc_fock_matrix(mol_h2o, Huv, eri, Duv)[0, 0] == \
+        pytest.approx(-32.57739541261037) and \
+        SCF.calc_fock_matrix(mol_h2o, Huv, eri, Duv)[2, 5] == \
+        pytest.approx(-1.6751501447185015) and \
+        SCF.calc_fock_matrix(mol_h2o, Huv, eri, Duv)[5, 2] == \
+        pytest.approx(-1.6751501447185015), \
+        "Fock Matrix Test (H2O) Failed"
+
+
+def test_solve_Roothan_equations(mol_h2o):
+    Duv = SCF.calc_initial_density(mol_h2o)
+    Fuv = SCF.calc_fock_matrix(mol_h2o, Huv, eri, Duv)
+    mo_e, mo_c = SCF.solve_Roothan_equations(Fuv, Suv)
+    assert mo_e == pytest.approx([-32.57830292,  -8.08153571,  -7.55008599,
+                                  -7.36396923,   -7.34714487,  -4.00229867,
+                                  -3.98111115]) \
+        and mo_c[0, :] == pytest.approx([-1.00154358e+00,  2.33624458e-01,

```

Comment:
This could potentially fail, as the sign in not guaranteed from run to run, so should be testing on absolute values.